stingray/powerspectrum.py

```diff
@@ -432,14 +432,14 @@ def __init__(self, lc, segment_size, norm="frac", gti=None):
         self.segment_size = segment_size
         self.norm = norm
         self.gti = gti
-        self.power_all, _ = AveragedPowerspectrum._make_segment_spectrum(
+        self.ps_all, _ = AveragedPowerspectrum._make_segment_spectrum(
             self, lc, segment_size)
-        self._make_matrix()
+        self._make_matrix(lc)
 
-    def _make_matrix(self):
+    def _make_matrix(self, lc):
         """Create the matrix with freq and time columns."""
-        row1 = self.power_all[0].freq
-        row2 = np.arange(0, self.segment_size*len(row1), self.segment_size)
+        row1 = self.ps_all[0].freq
+        row2 = np.arange(lc.time[0] - 0.5*lc.dt + 0.5*self.segment_size, lc.time[-1] + 0.5*lc.dt)
```

Comment:
`np.arange`  is missing the `step` argument (third argument), it should still be `self.segment_size`